src/link.js

```diff
@@ -34,3 +34,150 @@ export default class Link extends Plugin {
 		return 'Link';
 	}
 }
+
+/**
+ * The configuration of the {@link module:link/link~Link} feature.
+ *
+ * Read more in {@link module:link/link~LinkConfig}.
+ *
+ * @member {module:link/link~LinkConfig} module:core/editor/editorconfig~EditorConfig#link
+ */
+
+/**
+ * The configuration of the {@link module:link/link~Link link feature}.
+ *
+ *		ClassicEditor
+ *			.create( editorElement, {
+ * 				link:  ... // Link feature configuration.
+ *			} )
+ *			.then( ... )
+ *			.catch( ... );
+ *
+ * See {@link module:core/editor/editorconfig~EditorConfig all editor options}.
+ * @interface LinkConfig
+ */
+
+/**
+ * When set `true`, the `target="blank"` and `rel="noopener noreferrer"` attributes are automatically added to all external links
+ * in the editor. By external are meant all links in the editor content starting with `http`, `https`, or `//`.
+ *
+ * Internally, this option activates a predefined {@link module:link/link~LinkConfig#decorators automatic link decorator},
+ * which extends all external links with the `target` and `rel` attributes without additional configuration.
+ *
+ * **Note**: To control the `target` and `rel` attributes of specific links in the edited content, a dedicated
+ * {@link module:link/link~LinkDecoratorManualOption manual} decorator must be defined in the
+ * {@link module:link/link~LinkConfig#decorators `config.link.decodators`} array. In such scenario,
+ * the `config.link.targetDecorator` option should remain `undefined` or `false` to not interfere with the manual decorator.
+ *
+ * **Note**: It is possible to add other {@link module:link/link~LinkDecoratorAutomaticOption automatic}
+ * or {@link module:link/link~LinkDecoratorManualOption manual} link decorators when this option is active.
+ *
+ * More information about decorators can be found in the {@link module:link/link~LinkConfig#decorators decorators configuration}
+ * reference.
+ *
+ * @default false
+ * @member {Boolean} module:link/link~LinkConfig#targetDecorator
+ */
+
+/**
+ * Decorators provide an easy way to configure and manage additional link attributes in the editor content. There are
+ * two types of link decorators:
+ *
+ * * {@link module:link/link~LinkDecoratorAutomaticOption automatic} – they match links against pre–defined rules and
+ * manage their attributes based on the results,
+ * * {@link module:link/link~LinkDecoratorManualOption manual} – they allow users to control link attributes individually
+ *  using the editor UI.
+ *
+ * Link decorators are defined as an array of objects:
+ *
+ *		const linkConfig = {
```

Comment:
## Idea to discuss

Reading this piece of code an idea popped into my mind that it would be better if the decorators were named. 
I know this is shorter

```js
const linkConfig = {
	decorators: [
		{
			mode: 'automatic',
			callback: url => url.startsWith( 'http://' ),
			attributes: {
				target: '_blank',
				rel: 'noopener noreferrer'
			}
		},
		{
			mode: 'manual',
			label: 'Downloadable',
			attributes: {
				download: 'file.png',
			}
		},
		// ...
	]
}
```

but this would give us clearer attribute names in the model we can use in the command:

```js
const linkConfig = {
	decorators: [
		addTargetToExternals: {
			mode: 'automatic',
			callback: url => url.startsWith( 'http://' ),
			attributes: {
				target: '_blank',
				rel: 'noopener noreferrer'
			}
		},
		downloadableDecorator: {
			mode: 'manual',
			label: 'Downloadable',
			attributes: {
				download: 'file.png',
			}
		},
		// ...
	]
}
```

ATM there's no easy way for developers to figure out which decorator is responsible for what because they are numbered

![image](https://user-images.githubusercontent.com/1099479/59613999-2eb36300-9120-11e9-9558-d6badaa26889.png)

Reply:
I like the idea of having names for each decorator. 👍

Reply:
Currently I remain it untouched, as such modification will produce lot of docs changes. Let's w8 for more comments if we go into this way with configuration options.

Reply:
Named decorators sound like a good idea. Although I'm a bit worried about a possibility of attribute collision with existing model attr. Say someone names the decorator "bold", "linkHref" or something alike.

Prefixing them with `linkDecorator` should do the job.